### DISCUSSION TITLE
Pin_mapping.json Problem - Anpassung an Huawei R4850G2

### DISCUSSION THREAD

**Discussion by knopers1** [hoylabs/OpenDTU-OnBattery#1351] [General]
Guten Morgen, 
ich habe die Tage das Huawei R4850G2 bestellt, ein Gehäuse dazu gebaut, den MCP2515 CAN Modul sowie ein ein Kanal Relai beschafft..

Vorab wollte ich mich schon Mal um die Pin_mapping kümmern, damit ich die oben genannte Teile an den ESP32_S3 passend auf der Leiterplatte dran löten kann. Und jetzt kommt mein Problem:
Meine jetzige pin_mapping ohne Huawei sieht so aus:

![prob3](https://github.com/user-attachments/assets/bceb1291-6597-47fe-be5f-f9abefc2ca57)

**Ich habe kurz überschlagen, welche benutzbaren GPIO´s auf dem Esp32-s3 noch da sind und habe die pin:mapping ergänzt.**
},
        huawei": {
            "miso": 14,
            "mosi": 13,
            "clk": 26,
            "irq": 25,
            **"power": 33**, - GPIO33 gibt es nicht auf dem ESP32
            "cs": 15
 
Wenn ich die Pin-Mapping übertrage, bekomme ich folgende Fehlermeldung:

![prob1](https://github.com/user-attachments/assets/c95235bd-375f-4e3b-888c-c69427c87de7)

![prob2](https://github.com/user-attachments/assets/be75d841-8015-4947-8581-ca7d55f9d0bc)

Bis gestern noch, bin ich davon ausgeganngen, dass man beliebige GPIOs benutzen kann. Was bedeutet die Fehlermeldung genau? 
bei "Eth" kommt eine Änderung zustande, das abgeänderte Huawei bleibt bei -1 ....

Kann mir jemand erklären was ich falsch mache?

Schönen Sonntag 
Lukas

**Comment by Snusme83** [comment]
Servus, was steht in deiner pinmapping.json in Zeile 26, Spalte 9?

**Comment by knopers1** [comment]
ich gluabe den Fehler gefunden zu haben... Mein ESP32-s3 ist ein Dev-Kit-C.
Dort ist kein Gpio26 . Ich hatte wohl vor der Nase den falschen Pinout zu meinem ESP32 Board.

![DevKitC](https://github.com/user-attachments/assets/b9e5f23d-5050-46ae-8e6e-be562de2f551)

Ich habe direckt beim Huawei alles angepasst.

![prob1](https://github.com/user-attachments/assets/b3664598-7fa4-4026-9508-b60a419f131c)
![prob2](https://github.com/user-attachments/assets/7ba6e3d1-18f3-4b10-b30f-01f92014a919)


Ich versuche mein MCP2515 CAN Modul danach zu verbinden. Schauen wir mal, ob es funktioniert. Der Power: GPIO33 - bei mir jetzt 10 kommt auf das Relai.

**Comment by knopers1** [comment]
ich hatte leider keinen Erfolg. Das MPC2515 Joy-IT mit 16 Mhz Quarz verdrahtet exakt nach Pinout. 16 Mhz in der Einstellung gewählt, neustart .... nichts ! Kann es jemand bestätigen, dass der AC-Lader in Verbindung mit NRF für den HM-Hoymiles, Pylontech und ein Mal VE.Direkt unterstütz wird? Sonst muss etwas defekt sein ... Muss die CAN-Verbindung laufen, ohne dass eine Batterie auf der DC Seite angeschlossen ist? Ich hatte das Netzteil nur AC-Seitig am Strom…. Die CAN-Leitung ist ca. 2m lang. Der Abschlusswiederstand 120 Ohm war über den Jumper auf ON geschaltet. Das Relai über den Power Pin GPIO 10 hatte einmalig angezogen, so dass das Netzteil an war.

**Comment by p1ngb4ck** [comment]
Hi

Da das Thema so lang zurück liegt - erstmal die Frage - ist das noch aktuell für dich?
Für alle, die hier im Thema nach einer Antwort dazu suchen:

Ich habe es mit folgenden Settings hinbekommen:

ESP32-S3 N16R8 -> Das Ding mit Onboard-RGB-LED und 2 USB-C Ports.
SSD1306 - Display (irrelevant hier eigtl)
MCP2515 - das typische billig-Board, zB auf Amazon zu kriegen - modifiziert. 8MHz Crystal mit 16MHz Crystal ersetzt (hatte einen auf einem Spenderboard verfügbar) + 5V Leitung angelötet - ich teste noch ob es auch mit 8MHz und ohne Modifikation und mit LevelShifter klappt (würde löten ersparen)

Power-Pin geht via LevelShifter zu Relais, Relais versorgt die 5V Leitung des modifizierten Boards (bei unmodifiziertem Board und kompletter Verbindung via LevelShifter würde dann VCC komplett durchs Relais versorgt).

Pin-Mapping:
"huawei": {	
	"miso": 13,
	"mosi": 12,
	"clk": 11,
	"irq": 10,
	"power": 14,
	"cs": 15
}

Ach ja - mein eigenes R4950G2 ist noch nicht da - ich hab es verifiziert via:
- Hardware initialized im Log
- Arduino-Nano "CAN-Sniffer" zeigt die CAN-Requests vom OpenDTU im Serial-Log
- OpenDTU erhält ein ACK (keine Fehlermeldung mit fehlendem ACK mehr)

-> mal schauen, ob ich Zeit finde, einen Sketch zu bauen, der die "Emulation" eines R4850G2 ermöglicht -> Werte wären dann per Serial zu setzen -> einfach für Leute wie mich, die OpenDTU on battery bzw. ihren CAN Adapter testen wollen.
Grüße, Olli
